packages/core/test/resize-sensor/resizeSensorTests.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import { assert } from "chai";
+import { mount, ReactWrapper } from "enzyme";
+import * as React from "react";
+import { spy } from "sinon";
+
+import { IResizeSensorProps, ResizeSensor } from "../../src/components/resize-sensor/resizeSensor";
+
+describe("<ResizeSensor>", () => {
+    // this scope variable is assigned in mountResizeSensor() and used in resize()
+    let wrapper: ReactWrapper<IResizeTesterProps, any> | undefined;
+    const testsContainerElement = document.createElement("div");
+    document.documentElement.appendChild(testsContainerElement);
+
+    afterEach(() => {
+        // clean up list after each test, if it was used
+        if (wrapper !== undefined) {
+            wrapper.unmount();
+            wrapper.detach();
+            wrapper = undefined;
+        }
+    });
+    after(() => testsContainerElement.remove());
+
+    it("onResize is called when size changes", async () => {
+        const onResize = spy();
+        mountResizeSensor(onResize);
+        await resize({ width: 200 });
+        await resize({ width: 200 }); // this one is ignored
+        await resize({ height: 100 });
+        await resize({ width: 55 });
+        assert.equal(onResize.callCount, 3);
+        assertResizeArgs(onResize, ["200x0", "200x100", "55x100"]);
+    });
+
+    it("onResize is called when element changes", async () => {
+        const onResize = spy();
+        mountResizeSensor(onResize);
+        await resize({ width: 200, id: 1 });
+        await resize({ width: 200, id: 2 }); // not ignored bc element recreated
+        await resize({ width: 55, id: 3 });
+        assertResizeArgs(onResize, ["200x0", "200x0", "55x0"]);
+    });
+
+    it("onResize can be changed", async () => {
+        const onResize1 = spy();
+        mountResizeSensor(onResize1);
+        await resize({ width: 200, id: 1 });
+
+        const onResize2 = spy();
+        wrapper.setProps({ onResize: onResize2 });
+        await resize({ height: 100, id: 2 });
+        await resize({ width: 55, id: 3 });
+
+        assert.equal(onResize1.callCount, 1);
+        assert.equal(onResize2.callCount, 2);
+    });
+
+    function mountResizeSensor(onResize: IResizeSensorProps["onResize"]) {
+        return (wrapper = mount<IResizeTesterProps>(
+            <ResizeTester onResize={onResize} />,
+            // must be in the DOM for measurement
+            { attachTo: testsContainerElement },
+        ));
+    }
+
+    function resize(size: ISizeProps) {
+        wrapper.setProps(size);
+        return new Promise(resolve => setTimeout(resolve, 30));
```

Comment:
sooo resizes need time to propagate (more than a frame, typically), so we must wrap each one in this brief timeout. fortunately, `async/await` make this rather nice to use instead of callback hell.

Reply:
oh.... this seems like a pretty big gotcha

Reply:
if this is going to be relevant to users of this component, you should add a comment to the `onResize` prop.